mods/mc_worldmanager/realm/realmCategory.lua

```diff
@@ -0,0 +1,135 @@
+Realm.categories = {}
+
+function Realm.RegisterCategory(categoryDefinition)
+    Realm.categories[string.lower(categoryDefinition.key)] = categoryDefinition
+end
+
+function Realm:setCategoryKey(category)
+    self:set_data("category", category)
+end
+
+function Realm:getCategory()
+    local category = self:get_data("category")
+    if (category == nil or category == "") then
+        category = "default"
+    end
+
+    local categoryObject = Realm.categories[string.lower(category)]
+
+    return categoryObject
+end
+
+function Realm:AddOwner(ownerName)
+    local owners = self:get_data("owner")
+    if (owners == nil) then
+        owners = {}
+    end
+
+    owners[ownerName] = true
+    self:set_data("owner", owners)
+
+    if (areas ~= nil) then
+        self:AddPlayerAreaByName(ownerName)
+    end
+end
+
+function Realm:RemoveOwner(ownerName)
+    local owners = self:get_data("owner")
+    if (owners == nil) then
+        owners = {}
+    end
+
+    owners[ownerName] = nil
+    self:set_data("owner", owners)
+
+    if (areas ~= nil) then
+        self:RemovePlayerAreaByName(ownerName)
+    end
+end
+
+Realm.RegisterCategory({
+    key = "default",
+    visible = function(realm, player)
+        return true, "Default realms are visible to all players."
+    end,
+    joinable = function(realm, player)
+        return true, "Default realms are joinable by all players."
+    end
+})
+
+Realm.RegisterCategory({
+    key = "spawn",
+    visible = function(realm, player)
+        return true, "Spawn realms are visible to all players."
+    end,
+    joinable = function(realm, player)
+        return true, "Spawn realms are joinable by all players."
+    end
+})
+
+Realm.RegisterCategory({
+    key = "classroom",
+    visible = function(realm, player)
+
+        if (realm:get_data("students") == nil) then
+            realm:set_data("students", {})
+        end
+
+        if (realm:get_data("owner") == nil) then
+            realm:set_data("owner", {})
+        end
+
+        if (realm:get_data("students")[player:get_player_name()] ~= nil) then
```

Comment:
How are teachers treated? Need to consider the case that teachers should be able to join any realm (e.g., needing to help a student or grade an assignment/project), so perhaps they need to be added as owners?

Reply:
Users with the teachers privilege are assumed to be excluded from this check to make implementation details easier for any add-on mod. 

E.g., add-on mods should respect that teacher privileges override the check without creating a dependency.

See the teleportPlayer function in the realm player management file for an example of the assumption I'm making. 

I'll see if I can move this implementation from calling methods into this class.